### PR TITLE
[Docs] add Cloud details and additional guidance to Upgrade Strategies

### PR DESCRIPTION
**Pull request by mjseaman**
See this doc https://docs.google.com/document/d/19ch2mOhNl2Ig3O_Ld42A06rGmGCuxb-K7UiDIU060Gc/edit?tab=t.0 for context.

**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by jtvdez — APPROVED** [review]

**Review by jtvdez — APPROVED** [review]

**Comment by mjseaman** [comment]
@timlevett it looks like I can't merge into main. Can you?

File: docs/sources/upgrade-guide/when-to-upgrade/index.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ We no longer wait for the yearly major release to give you access to the next bi
 
 Having a dependable release process provides users like you with the best Grafana experience possible, and it provides the flexibility to upgrade in a manner that works best for you and your organization.
 
+{{< admonition type="note" >}}
+Grafana Cloud follows a different release cadence than Grafana OSS and Enterprise. To learn more about Cloud release channels, refer to [Rolling release channels for Grafana Cloud](https://grafana.com/docs/rolling-release/).
+{{< /admonition >}}
```
Comment (timlevett) [inline docs/sources/upgrade-guide/when-to-upgrade/index.md:23]: Tricky to word here, but is it okay that we are generalizing "Grafana Cloud", the whole suite is not on RRC, just Grafana Core

```suggestion
{{< admonition type="note" >}}
Grafana Cloud follows a different release cadence than Grafana OSS and Enterprise. Grafana Cloud, much like many other SaaS product, has different deployment cadences for various parts of the application suite. Grafana Core in Grafana Cloud uses Rolling release channels. To learn more about release channels, refer to [Rolling release channels for Grafana Cloud](https://grafana.com/docs/rolling-release/).
{{< /admonition >}}
```

File: docs/sources/upgrade-guide/when-to-upgrade/index.md [hunk 2]
```diff
@@ -138,12 +142,45 @@ A critical feature degradation usually meets one of the following criteria:
...
+
+Reduce the risk of bugs by staying current and rolling out upgrades across dev or test environments before production. A Grafana Enterprise license entitles you to an additional dev and test instance for this purpose, available through your account team, and in Grafana Cloud you can create dev and test stacks that upgrade before production by using [rolling release channels](https://grafana.com/docs/rolling-release/).
+
+- Roll back and report issues if you experience problems in dev or test.
+- In Cloud, run a dev or test stack on the `fast` channel, and run your production stack on `steady` or `slow`.
+- To change your release channel, open a support ticket.
```
Comment (timlevett) [inline docs/sources/upgrade-guide/when-to-upgrade/index.md:174]:
```suggestion
- Take a backup of your database before you do an upgrade. Best practices suggest to take a backup of your database on a regular basis.
- Roll back and report issues if you experience problems in dev or test.
- In Cloud, run a dev or test stack on the `fast` channel, and run your production stack on `steady` or `slow`.
- To change your release channel, open a support ticket.
```

File: docs/sources/upgrade-guide/when-to-upgrade/index.md [hunk 3]
```diff
@@ -138,12 +142,45 @@ A critical feature degradation usually meets one of the following criteria:
...
+
+### Known breaking changes
+
+As a rule we always seek backward compatibility and migration, and reserve breaking changes for Grafana’s once-yearly major release. However occasionally small breaking changes (like updates to API payloads) will ship in minor releases. These are announced in [upgrade guides](https://grafana.com/docs/grafana/latest/upgrade-guide/), [What’s New](https://grafana.com/docs/grafana-cloud/whats-new/), and our [changelog](https://github.com/grafana/grafana/blob/main/CHANGELOG.md).
+
+Always read the [upgrade guide](https://grafana.com/docs/grafana/latest/upgrade-guide/) and [changelog](https://github.com/grafana/grafana/blob/main/CHANGELOG.md) prior to upgrading to learn about and account for breaking changes.
```
Comment (timlevett) [inline docs/sources/upgrade-guide/when-to-upgrade/index.md:180]:
```suggestion
Always read the [upgrade guide](https://grafana.com/docs/grafana/latest/upgrade-guide/) and [changelog](https://github.com/grafana/grafana/blob/main/CHANGELOG.md) prior to upgrading to learn about and account for breaking changes.

For Grafana Cloud, we will take care of any known breaking changes for you, we will have a migration strategy and/or inform you if we are removing features. To learn more about this please read our deprecation policy.
```

**Review by timlevett — COMMENTED** [review]
Overall looks good, added a few items after reading it over again.

File: docs/sources/upgrade-guide/when-to-upgrade/index.md (hunk 3 above)
Comment (mjseaman) [reply docs/sources/upgrade-guide/when-to-upgrade/index.md:180]: @timlevett is there a link to our deprecation policy?

Comment (timlevett) [reply docs/sources/upgrade-guide/when-to-upgrade/index.md:180]: Best we have publicly is https://github.com/grafana/grafana/blob/main/contribute/breaking-changes-guide/breaking-changes-guide.md#i-know-its-a-breaking-change-whats-next i believe

**Review by timlevett — APPROVED** [review]

**Pull request merged**